icu4c/source/common/ucurr.cpp

```diff
@@ -967,22 +974,30 @@ collectCurrencyNames(const char* locale,
     CharString loc = ulocimp_getName(locale, ec2);
     if (U_FAILURE(ec2)) {
         ec = U_ILLEGAL_ARGUMENT_ERROR;
+        return;
     }
 
     // Get maximum currency name count first.
     getCurrencyNameCount(loc.data(), total_currency_name_count, total_currency_symbol_count);
 
+    *currencySymbols = nullptr;
```

Comment:
I'd suggest that you move this statement to inside of the if-statement, together with setting `*total_currency_name_count` and `*total_currency_symbol_count`, that would be more logical grouping.